modules/features2d/src/matchers.cpp

```diff
@@ -601,22 +601,32 @@ void DescriptorMatcher::checkMasks( InputArrayOfArrays _masks, int queryDescript
 {
     std::vector<Mat> masks;
     _masks.getMatVector(masks);
-
+    int rows;
     if( isMaskSupported() && !masks.empty() )
     {
         // Check masks
         size_t imageCount = std::max(trainDescCollection.size(), utrainDescCollection.size() );
         CV_Assert( masks.size() == imageCount );
         for( size_t i = 0; i < imageCount; i++ )
-        {
-            if( !masks[i].empty() && (!trainDescCollection[i].empty() || !utrainDescCollection[i].empty() ) )
-            {
-                int rows = trainDescCollection[i].empty() ? utrainDescCollection[i].rows : trainDescCollection[i].rows;
+        { 
+            if(!masks[i].empty()){
+        	  if(!trainDescCollection[i].empty() || !utrainDescCollection[i].empty())
+        	  { 
+                    rows = trainDescCollection[i].rows;
```

Comment:
Execution will fall into this branch if either trainDescCollection or utrainDescCollection isn't empty. So rows should be populated in accordance.